File: tests/ruby/service.rb
require 'json'
require 'httparty'

class UserService
  def initialize
    @users = {}
  end

  def create_user(name)
    id = @users.length + 1
    user = { id: id, name: name }
    @users[id] = user
    user
  end

  def get_user(id)
    @users[id]
  end

  def delete_user(id)
    @users.delete(id)
  end
end

class AdminService < UserService
  def create_admin(name, role)
    user = create_user(name)
    user[:role] = role
    user
  end
end

def helper_function
  'helper'
end

File: tests/php/Service.php
<?php

class UserService {
    private array $users = [];

    public function createUser(string $name): array {
        $id = count($this->users) + 1;
        $user = ['id' => $id, 'name' => $name];
        $this->users[$id] = $user;
        return $user;
    }

    public function getUser(int $id): ?array {
        return $this->users[$id] ?? null;
    }

    public function deleteUser(int $id): void {
        unset($this->users[$id]);
    }
}

class AdminService extends UserService {
    public function createAdmin(string $name, string $role): array {
        $user = $this->createUser($name);
        $user['role'] = $role;
        return $user;
    }
}

function helper_function(): string {
    return 'helper';
}
